client/python/openlineage/client/transport/kafka.py

```diff
@@ -57,18 +61,39 @@ class KafkaTransport(Transport):
     def __init__(self, config: KafkaConfig) -> None:
         self.topic = config.topic
         self.flush = config.flush
+        self.message_key = config.messageKey
         self.kafka_config = config
         self._is_airflow_sqlalchemy = _check_if_airflow_sqlalchemy_context()
         self.producer = None
         if not self._is_airflow_sqlalchemy:
             self._setup_producer(self.kafka_config.config)
         log.debug("Constructing openlineage client to send events to topic %s", config.topic)
 
-    def emit(self, event: Union[RunEvent, DatasetEvent, JobEvent]) -> None:  # noqa: UP007
+    def _get_message_key(self, event: RunEvent | DatasetEvent | JobEvent) -> str:
+        if isinstance(event, DatasetEvent):
+            return f"dataset:{event.dataset.namespace}/{event.dataset.name}"
+
+        if isinstance(event, JobEvent):
+            return f"job:{event.job.namespace}/{event.job.name}"
```

Comment:
Currently Transport interface in Java client does not allow passing DatasetEvent and JobEvent to `.emit()` method. I'll open a separated PR for that